src/vcpkg/base/files.cpp

```diff
@@ -2898,6 +2898,8 @@ namespace vcpkg
             return true;
 #else // ^^^ (defined(__APPLE__) // !defined(__APPLE__) vvv
 
+            destination_fd.fchmod(source_stat.st_mode, ec);
```

Comment:
This is now generally *before* writing the output and for some paths before reading `ec`, instead of immediately before returning. IMO this is a behavioural change, not deduplication.

Reply:
Perhaps, but is there a need to delay the fchmod until the end, requiring the duplication?

Reply:
Maybe it is really an improvement ;-)

Reply:
I think this change LGTM, but it needs to add a check for `ec` and an early failure if the `fchmod` failed.

Reply:
Thanks that's a good point, I pushed a change.